### PR TITLE
Correct behavior of owl_get_station

### PR DESCRIPTION
**Pull request by willwillhi1**
In the implementation of owl_dump_station(), I found that the behavior of owl_get_station() was incorrect, so I correct the behavior of owl_get_station() and add the function of owl_dump_station().

### CODE REVIEW

**Review by jserv — CHANGES_REQUESTED** [review]
Emphasize on the fixes for owl_get_station() in this pull request.
For implementing dump_station(), please work on #36 .

**Review by jserv — CHANGES_REQUESTED** [review]
Read CONTRIBUTING.md carefully for consistent coding style.

File: vwifi.c [hunk 1]
```diff
@@ -720,7 +720,23 @@ static int owl_get_station(struct wiphy *wiphy,
 {
     struct owl_vif *vif = ndev_get_owl_vif(dev);
 
-    if (memcmp(mac, vif->bssid, ETH_ALEN))
+    bool found_sta = false;
+    if (dev->ieee80211_ptr->iftype == NL80211_IFTYPE_AP) {
```
Comment (jserv) [inline vwifi.c:724]: Rewrite `if` and `else if` with switch-case.

**Review by rickywu0421 — APPROVED** [review]

**Comment by jserv** [comment]
Thank @willwillhi1 for contributing!

**Pull request merged**